src/cargo/ops/cargo_add/mod.rs

```diff
@@ -104,6 +105,14 @@ pub fn add(workspace: &Workspace<'_>, options: &AddOptions<'_>) -> CargoResult<(
                     options.gctx,
                     &mut registry,
                 )
+                .map_err(|err| {
+                    if err.is::<NameValidationError>() {
```

Comment:
Thanks for limiting this to errors rather than making assumptions about the naming rules which set a trap for us if we changed those rules in the futured